src/core_plugins/tile_map/public/maps_visualization.js

```diff
@@ -272,10 +272,40 @@ export function MapsVisualizationProvider(serviceSettings, Notifier, getAppState
 
       const indexPatternName = agg.vis.indexPattern.id;
       const field = agg.fieldName();
-      const filter = { meta: { negate: false, index: indexPatternName } };
-      filter[filterName] = { ignore_unmapped: true };
-      filter[filterName][field] = filterData;
-      getAppState().$newFilters = [filter];
+      const query = this.vis.API.queryManager.getQuery();
+      const language = query.language;
+
+      if (language === 'lucene') {
+        const filter = { meta: { negate: false, index: indexPatternName } };
+        filter[filterName] = { ignore_unmapped: true };
+        filter[filterName][field] = filterData;
+
+        this.vis.API.queryFilter.addFilters([filter]);
+      }
+      else if (language === 'kuery') {
+        const { fromKueryExpression, toKueryExpression, nodeTypes } = this.vis.API.kuery;
+        let newQuery;
+
+        if (filterName === 'geo_bounding_box') {
```

Comment:
could we have a simple abstraction on top of this, like `this.vis.API.queryManager.buildQuery(filterName, filterParams)` ?

Reply:
There isn't a one to one relationship between ES filters and Kuery queries so such an abstraction isn't possible. The fact that this file even uses ES filter names couples it to the ES query DSL which isn't good. We'll eventually want to get rid of the ES filter names completely.